modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java

```diff
@@ -858,7 +858,12 @@ private void storeEntries(Collection<IgniteBiTuple<ByteArray, byte[]>> entries,
 
                     entries.forEach(e -> batch.put(e.getKey(), e.getValue()));
 
-                    return vaultMgr.putAll(batch);
+                    // skip already processed revision
```

Comment:
Seems that it's also not required within given solution because all watches are registered on start only.